aioshelly/__init__.py

```diff
@@ -162,8 +163,9 @@ async def create(
     ):
         """Device creation."""
         if isinstance(ip_or_options, str):
-            options = ConnectionOptions(ip_or_options)
+            options = ConnectionOptions(gethostbyname(ip_or_options))
```

Comment:
Is this doing I/O ? 

Also, should it be done in this method, or done when we create a coap context which actually needs it?

Reply:
> Is this doing I/O ?

Hmm I think, yes. I'm not sure what to do with that. Call it in the event loop?

> Also, should it be done in this method, or done when we create a coap context which actually needs it?

If we resolve hostname to IP only in `COAP.subscribe_updates()` then the aioshelly device `ip_address` property will contain the hostname, not IP. We can use `gethostbyname` in `Device.__init__()`
```python
class Device:
    """Shelly device reppresentation."""

    def __init__(
        self,
        coap_context: COAP,
        aiohttp_session: aiohttp.ClientSession,
        options: ConnectionOptions,
    ):
        options.ip_address = gethostbyname(options.ip_address)
        self.coap_context = coap_context
        self.aiohttp_session = aiohttp_session
        self.options = options
```

Reply:
Well first find out if it does I/O. If it does, it should run in the executor.

Reply:
`gethostbyname` does I/O. Can we proceed with this change?